test/metabase/test/data/sparksql.clj

```diff
@@ -0,0 +1,120 @@
+(ns metabase.test.data.sparksql
+  (:require [clojure.java.jdbc :as jdbc]
+            [clojure.string :as s]
+            [environ.core :refer [env]]
+            (honeysql [core :as hsql]
+                      [format :as hformat]
+                      [helpers :as h])
+            (metabase.driver [generic-sql :as sql]
+                             [hive-like :as hive-like])
+            (metabase.test.data [generic-sql :as generic]
+                                [interface :as i])
+            [metabase.util :as u]
+            [metabase.util.honeysql-extensions :as hx])
+  (:import metabase.driver.sparksql.SparkSQLDriver))
+
+(def ^:const field-base-type->sql-type
+  {:type/BigInteger "BIGINT"
+   :type/Boolean    "BOOLEAN"
+   :type/Date       "DATE"
+   :type/DateTime   "TIMESTAMP"
+   :type/Decimal    "DECIMAL"
+   :type/Float      "DOUBLE"
+   :type/Integer    "INTEGER"
+   :type/Text       "STRING"})
+
+(defn quote-name [nm]
+  (str \` nm \`))
+
+(defn- dash-to-underscore [s]
+  (s/replace s #"-" "_"))
+
+(defn- qualified-name-components
+  ([db-name]                       (map dash-to-underscore [db-name]))
+  ([db-name table-name]            (map dash-to-underscore [db-name table-name]))
+  ([db-name table-name field-name] (map dash-to-underscore [table-name field-name])))
+
+(defn database->connection-details [context {:keys [database-name]}]
+  (merge {:host "localhost"
+          :port 10000
+          :user "admin"
+          :password "admin"}
+         (when (= context :db)
+           {:db (dash-to-underscore database-name)})))
+
+(defn- do-insert!
+  "Insert ROWS-OR-ROWS into TABLE-NAME for the DRIVER database defined by SPEC."
+  [driver spec table-name row-or-rows]
+  (let [prepare-key (comp keyword (partial generic/prepare-identifier driver) name)
+        rows        (if (sequential? row-or-rows)
+                      row-or-rows
+                      [row-or-rows])
+        columns     (keys (first rows))
+        values      (for [row rows]
+                      (for [value (map row columns)]
+                        (sql/prepare-value driver {:value value})))
+        hsql-form   (-> (h/insert-into (prepare-key table-name))
+                        (h/values values))
+        sql+args    (hive-like/unprepare
+                     (hx/unescape-dots (binding [hformat/*subquery?* false]
+                                         (hsql/format hsql-form
+                                                      :quoting             (sql/quote-style driver)
+                                                      :allow-dashed-names? false))))]
+    (with-open [conn (jdbc/get-connection spec)]
+      (try
+        (do
+          (.setAutoCommit conn false)
+          (jdbc/execute! {:connection conn} sql+args {:transaction? false}))
```

Comment:
You don't need to wrap multiple statements in a `try` form inside a `do` form.

You can just do

```clojure
(try
  (.setAutoCommit conn false)
  (jdbc/execute! {:connection conn} sql+args {:transaction? false}))
  (catch ...))
```